parfive/downloader.py

```diff
@@ -319,7 +319,9 @@ def _format_results(self, retvals, main_pb):
             elif isinstance(res, Exception):
                 raise res
             else:
-                results.append(res)
+                requested_url, filepath = res
+                results.append(filepath)
+                results.urls.append(requested_url)
```

Comment:
Perhaps we should have an append method on results which takes both a filepath and a url?

Reply:
Seems reasonable but I think a larger refactor of the `Results` object will be needed soon as `res = res1 + res2` works for the path but not the error and now the urls either same for other operation supported by UserList but not the results object

Reply:
So I've implemented a basic version here and created new issue around the limitations #152